project/webapp/controller/ProductsList.controller.js

```diff
@@ -264,6 +296,45 @@ sap.ui.define(
             [bGreaterThanOne ? aSelectedItems.length : sProductName]
           );
         },
+
+        _getGroups(oContext) {
+          const rating = oContext.getProperty("rating");
+
+          switch (rating) {
```

Comment:
too many repeated code
why don't you write:
return {
                key: `rating${rating}`,
                text: this.oResourceBundle.getText("GroupRatingText", [rating])
              }